### PR TITLE
Allow USB mode selection on USB connect

### PR DESCRIPTION
**Pull request by schwabe**

### CODE REVIEW

File: radio/src/targets/common/arm/stm32/CMakeLists.txt
Comment (projectkk2glider) [inline radio/src/targets/common/arm/stm32/CMakeLists.txt]: Indentation

File: radio/src/debug.h
Comment (projectkk2glider) [inline radio/src/debug.h]: This looks wrong. The idea was/is:
 * in USB serial mode: call `serialPrintf` only if `cliTracesEnabled == true`
 * in real serial port mode: call `serialPrintf` if `(serial2TracesEnabled() && cliTracesEnabled)`

File: radio/src/gui/128x64/radio_setup.cpp
Comment (bsongis) [inline radio/src/gui/128x64/radio_setup.cpp]: perhaps an enum instead of 3 ?

File: radio/src/targets/common/arm/stm32/serial2_driver.cpp
Comment (bsongis) [inline radio/src/targets/common/arm/stm32/serial2_driver.cpp]: indentation issue here

File: radio/src/targets/common/arm/stm32/usb_driver.h
Comment (bsongis) [inline radio/src/targets/common/arm/stm32/usb_driver.h:29]: here is a nice enum which needs to be used in the menus!

File: radio/src/targets/common/arm/stm32/usbd_desc.c
Comment (bsongis) [inline radio/src/targets/common/arm/stm32/usbd_desc.c]: isn't it possible to include board.h without the path here?

**Review by bsongis — APPROVED** [review]
Excellent PR!

**Comment by ReFil** [comment]
soooo how do we use this? i just installed @sirPerna 's built version of opentx for the baudrate switch, and it just goes to usb drive not simulator mode when i plug it in when powered on.

**Comment by schwabe** [comment]
@livid-fpv I have no idea how @sirPerna version was built. If you don't have the option to select usb mode in general settings, his build was done before this was merged.

Keep an eye on the downloads page we should release nightly builds soon.

**Comment by kilrah** [comment]
> soooo how do we use this?

You wait for the release of OpenTX 2.2.1.

**Comment by ReFil** [comment]
ok @schwabe  It was build on the 10th. Thanks for the information

**Pull request merged**
